src/Parser/Desugar.ml

```diff
@@ -708,7 +708,11 @@ and tr_def ?(public=false) (def : Raw.def) =
     let (pub_type, pub_ctors) =
       match vis with
       | DV_Private  -> (public, public)
-      | DV_Abstract -> (true,   public)
+      | DV_Abstract -> if public then
+                       (InterpLib.Error.report ~pos:def.pos
```

Comment:
`InterpLib.Error.report` is too low-level to use it here. We should use `Error.warn` and keep all message strings in one place (`Error.ml`). Of course, a new kind of warning should be added to `Error` module.